### DISCUSSION TITLE
Is it possible to skip an entire PhaseGroup based on previous tests ?

### DISCUSSION THREAD

**Discussion by tverdonck** [google/openhtf#1081] [Q&A] [answered]
Hi,

Is it possible to skip an entire PhaseGroup based on previous tests ? 

I know test flow execution can be altered using PhaseOption.run_if decorators but they have no context.

Is it possible to alter test flow execution based on 'test state' ?

With Subtest and PhaseResult.FAIL_SUBTEST one can skip all remaining 'tests' from the Subtest but this results in a FAIL.

If possible; an example whould be greatly appreciated

Regards,
T

**Comment by arsharma1** [comment, marked as answer]
This is possible. Please see https://github.com/google/openhtf/blob/master/openhtf/core/phase_branches.py#L207. It will require wrapping your PhaseGroup in a Subtest.

**Reply by tverdonck** [reply]
Thank you for the quick response.

Could you give me an example please.  

Does this also required a FAIL condition (like Subtest withPhaseResult.FAIL_SUBTEST) ?

Regards,
Tom
